### ISSUE TITLE
GD diverges when template is used out of the box from BenchOpt tutorial

### ISSUE THREAD

**Issue by jflamant** [benchopt/template_benchmark#49]
Hello ! 

While running [this tutorial](https://benchopt.github.io/stable/tutorials/build_benchmark.html), cloning the BenchOpt template and running `benchopt run my_benchmark` leads to diverging behavior of the GD solver. Hence no output is produced. In the tutorial, it is mentioned that 

>  The solver will use a predefined set of stepsizes, in our case [1, 1.99] (the stepsize is scaled by the inverse of the gradient’s Lipschitz constant, you can ignore this detail if you are not familiar with this concept).

However this does not appear to be the case in the current version of template. Updating the method `run()` in  ``python-gd.py`` as 
```
def run(self, callback):

        self.beta = np.zeros(self.X.shape[1])
        L = np.linalg.svdvals(self.X)[0]**2 # compute gradient's Lipschitz constant
        while callback():
            self.beta -= (self.learning_rate / L) * gradient_ols(
                self.X, self.y, self.beta
            )
```
did work for me. Is this a bug and if so, would a PR be meaningful? 

Thanks!

**Comment by mathurinm** [comment]
Thanks @jflamant, indeed the current scaling by the inverse gradient Lipschitz constant is missing. A PR would be great!

**Comment by jflamant** [comment]
Thanks @mathurinm ! I will try to propose a PR by tonight then.
